amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/RecordValidatorQueue.java

```diff
@@ -0,0 +1,66 @@
+package software.amazon.kinesis.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import java.util.*;
+
+@Slf4j
+public class RecordValidatorQueue {
+
+    HashMap<String, List<String>> dict = new HashMap<>();
+
+    public void add( String shardId, String data ) {
+        if ( dict.containsKey( shardId ) ) {
+            // Just add the data to this item
+            List<String> oldVal = dict.get( shardId );
+            oldVal.add( data );
+            dict.put( shardId, oldVal );
+        } else {
+            List<String> newVal = new ArrayList<>();
+            newVal.add( data );
+            dict.put( shardId, newVal );
+        }
+    }
+
+    public int validateRecords( int trueTotalShardCount ) {
+        // Validate that each List in the HashMap has data records in increasing order
+
+        boolean incOrder = true;
+        for ( Map.Entry<String, List<String>> entry : dict.entrySet() ) {
+            List<String> recordsPerShard = entry.getValue();
+            int prevVal = -1;
+            boolean shardIncOrder = true;
+            for ( String record : recordsPerShard ) {
+                int nextVal = Integer.parseInt( record );
```

Comment:
I make this assumption because I am working on the assumption that all the data I am putting onto streams in these integration tests are incrementing numbers "1", "2", "3", ... as BigIntegers. In TestRecordProcessorV2/processRecords, the data is read and converted from a byte array to a String. I could do the parsing here, but that seems to be to be making the same assumption that you point out here. Is there any way that I can improve this or note the assumption that all the data on these streams are integers?